### PR TITLE
dnn: update network dump code, include ngraph serialization

### PR DESCRIPTION
**Pull request by alalek**
relates #17382

<cut/>

```
force_builders=Custom,Custom Win,Custom Mac
build_image:Custom=ubuntu-openvino-2020.2.0:16.04
build_image:Custom Win=openvino-2020.2.0
build_image:Custom Mac=openvino-2020.2.0

test_modules:Custom=dnn,python2,python3,java
test_modules:Custom Win=dnn,python2,python3,java
test_modules:Custom Mac=dnn,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

File: modules/dnn/src/ie_ngraph.cpp [hunk 1]
```diff
@@ -440,9 +448,27 @@ void InfEngineNgraphNet::init(Target targetId)
             ngraph_function->validate_nodes_and_infer_types();
         }
         cnn = InferenceEngine::CNNNetwork(ngraph_function);
-#ifdef _DEBUG  // TODO
-        //cnn.serialize("/tmp/cnn.xml", "/tmp/cnn.bin");
+
+        if (DNN_IE_SERIALIZE)
+        {
+#ifndef OPENCV_DNN_DISABLE_NETWORK_AUTO_DUMP
```
Comment (dkurt) [inline modules/dnn/src/ie_ngraph.cpp:454]: Where this is defined? And what is default value?

Comment (alalek) [reply modules/dnn/src/ie_ngraph.cpp:454]: It is not defined anywhere in OpenCV, but can be passed through build flags.
Purpose is custom builds which want to turn off network dumps.

Comment (dkurt) [reply modules/dnn/src/ie_ngraph.cpp:454]: I just though that `AUTO_DUMP` means to dump networks by default, without flags. If it's build flag, that means that if `OPENCV_DNN_DISABLE_NETWORK_AUTO_DUMP` is not specified but `OPENCV_DNN_IE_SERIALIZE` is ON we cannot dump the network?

File: modules/dnn/src/ie_ngraph.cpp [hunk 2]
```diff
@@ -440,9 +448,27 @@ void InfEngineNgraphNet::init(Target targetId)
...
+            std::string dumpFileNameBase = netImpl_.getDumpFileNameBase();
+            try
+            {
+                cnn.serialize(dumpFileNameBase + "_ngraph.xml", dumpFileNameBase + "_ngraph.bin");
```
[the remark this answers is not in the document]
Comment (dkurt) [reply modules/dnn/src/ie_ngraph.cpp:458]: Looks like a bug in Inference Engine. Have you checked an accuracy after a single `serialize`? Maybe it breaks even inference.

File: modules/dnn/src/ie_ngraph.cpp (hunk 1 above)
Comment (alalek) [reply modules/dnn/src/ie_ngraph.cpp:454]: Naming is aligned with [this code](https://github.com/opencv/opencv/blame/master/modules/dnn/src/dnn.cpp#L3416-L3428).

Idea is compile-time blocker to dump/serialize network.

Comment (dkurt) [reply modules/dnn/src/ie_ngraph.cpp:454]: So, basically my motivation is to have this option available in OpenCV from OpenVINO distribution. Sometimes to enable models in OpenVINO it's helpful to have this method. Will this build flag be available in OpenVINO build by default or we can miss it?

Comment (alalek) [reply modules/dnn/src/ie_ngraph.cpp:454]: This code is enabled by default, including OpenVINO builds.

Comment (dkurt) [reply modules/dnn/src/ie_ngraph.cpp:454]: Oh, I see now, sorry. `#ifndef` and `DISABLE`. Thanks!

**Comment by dkurt** [comment]
Faced the following error:
```
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(3.4.10-dev) /home/dkurt/opencv/modules/core/src/system.cpp:2026: error: (-5:Bad argument) Invalid value for parameter OPENCV_DNN_IE_SERIALIZE: ./model.xml in function 'read'

Aborted (core dumped)
```
when
```
export OPENCV_DNN_IE_SERIALIZE=./model.xml
```

**Pull request merged**